[serf/subversion] Add subversion and serf ports

rhuijben opened the pull request:
Adds support for building the subversion client libraries.

Do this by adding Apache Serf and Apache Subversion ports.

The ports should work on:
* x86-windows
* x64-windows
* x86-windows-static-md
* x64-windows-static-md

The patches applied for Subversion are already in the base Subversion repository, but will most likely only be included in the next feature release (=1.x release).

rhuijben committed:
[subversion] Add new Subversion port, with required serf dependency

Neumann-A commented on ports/serf/portfile.cmake:
```diff
@@ -0,0 +1,116 @@
+    SHA512 9f5418d991840a08d293d1ecba70cd9534a207696d002f22dbe62354e7b005955112a0d144a76c89c7f7ad3b4c882e54974441fafa0c09c4aa25c49c021ca75d
+)
+
+vcpkg_download_distfile(SCONS_ARCHIVE
+    URLS "https://prdownloads.sourceforge.net/scons/scons-local-${SCONS_VERSION}.tar.gz"
+    FILENAME "scons-local-${SCONS_VERSION}.tar.gz"
+    SHA512 e52da08dbd4e451ec76f0b45547880a3e2fbd19b07a16b1cc9c0525f09211162ef59ff1ff49fd76d323e1e2a612ebbafb646710339569677e14193d49c8ebaeb
+)
```
`vcpkg_find_acquire_program(SCONS)` ?

rhuijben replied:
Don't think that will work yet. Couldn't find a single reference to scons in the build tree.

This works for the current serf version, and the next one (if ever released as stable) switches to cmake.

Neumann-A replied:
https://github.com/microsoft/vcpkg/blob/fe112d8301e13fde9a25dc92d94a0c4a6dc3cf19/scripts/cmake/vcpkg_find_acquire_program.cmake#L355-L363

https://github.com/microsoft/vcpkg/blob/fe112d8301e13fde9a25dc92d94a0c4a6dc3cf19/scripts/cmake/vcpkg_find_acquire_program.cmake#L15-L39

rhuijben replied:
Thanks... I found it and tried it... and determined that it doesn't work. 

This version of scons is incompatible with Visual Studio 2019 without further patches.

I think this should be updated as a separate patch. (Might need more work as just bumping versions and download references didn't work either).

For now I'm trying to get my existing legacy NAnt scripting converted to VCPkg to enable the use of github workflows for a few depending projects.

Neumann-A commented on ports/serf/portfile.cmake:
```diff
@@ -0,0 +1,116 @@
+  set(EXTRA_MODE ${EXTRA_MODE} STATIC_CRT=yes)
+endif()
+
+vcpkg_execute_build_process(
+    COMMAND ${PYTHON3}
+        ${SCONS_PATH}/scons.py
+            PREFIX=${CURRENT_PACKAGES_DIR}
+            LIBDIR=${CURRENT_PACKAGES_DIR}/lib
+            OPENSSL=${CURRENT_INSTALLED_DIR}
+            ZLIB=${CURRENT_INSTALLED_DIR}
+            APR=${CURRENT_INSTALLED_DIR}
+            APU=${CURRENT_INSTALLED_DIR}
+            SOURCE_LAYOUT=no
+            ${EXTRA_MODE}
+            ${SCONS_ARCH}
+            DEBUG=no
+            install-lib install-inc
+    WORKING_DIRECTORY ${SOURCE_PATH}
+)
```
this needs to setup build flags from the cmake toolchain

rhuijben replied:
Would be nice, yes... But I don't think APR and anything below it really supports anything but 100% Shared libraries, or 100% static. APR is like the CRT for the projects depending on it. You can't mix memory management.

Neumann-A commented on ports/subversion/portfile.cmake:
```diff
@@ -0,0 +1,128 @@
+  message(FATAL_ERROR "${PORT} does not currently support this platform")
+endif()
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
+  set(BUILD_MODE --with-shared-serf)
+else()
+  set(BUILD_MODE --with-static-apr --with-static-openssl --disable-shared)
+endif()
```
this is wrong for custom triplets. probably requires #17029 @ras0219-msft

rhuijben replied:
Subversion has its own buildfile generator. This produces proper msbuild files, and the settings are then overloaded bij the vcproj msbuild runner.

Neumann-A replied:
You are assuming here that `VCPKG_LIBRARY_LINKAGE` is constant for all ports and thus setting 
 `--with-shared-serf`
or
`--with-static-apr --with-static-openssl`
without knowing if serf, apr or openssl have really been built with the same `VCPKG_LIBRARY_LINKAGE`. Please read the vcpkg docs about per port customization (https://github.com/microsoft/vcpkg/blob/master/docs/users/triplets.md#per-port-customization).

rhuijben replied:
This does only change the name of the .lib linked against, as generated in the .vcxproj files later build with msbuild... It doesn't build apr itself, not any of these other dependencies.

But I do want apr-1.lib (static);vs libapr-1.lib (shared) to be correct.

Note I maintained quite a bit of the Windows specific scripting on the Subversion side over the last decade and a half. You should really check the code there before blocking a patch for the wrong reasons.

The resulting binaries through these patches have been tested by the Subversion project.

But Subversion and most of its ASF dependencies don't have a native static+static crt build, so this specific scenario needs separate patches.

Neumann-A replied:
> But I do want apr-1.lib (static);vs libapr-1.lib (shared) to be correct.

Consider the triplet:
```
set(VCPKG_LIBRARY_LINKAGE dynamic)
if(PORT MATCHES "apr")
    set(VCPKG_LIBRARY_LINKAGE static)
endif()
```

how does it affect the build? Is the portfile correct in this case? Can it still find the library?

> Note I maintained quite a bit of the Windows specific scripting on the Subversion side over the last decade and a half. You should really check the code there before blocking a patch for the wrong reasons.

Don't care about that. I just care about correctness and none of your "reasons". `VCPKG_LIBRARY_LINKAGE` is a per port value and can be different on a per port basis. So consider answering my concerns instead of trying to use your "position" to convince me. 

Furthermore I am not blocking, I am just trying to get something like #17029 into vcpkg because it is sometimes needed/required by ports how dependent ports are build.

JackBoosY replied:
I think what Neumann said is reasonable, this will pass our official triad, but not unofficially.

rhuijben replied:
@JackBoosY This will ask changes to at least quite a few dependencies before leaf ports can use this. Libraries like apr don't support al these scenarios as they provide memory management to their dependencies and you can't just mix CRT versions like that or things will break in the most unexpected locations.

Serf and Subversion depend on the apr memory management for their internals, so I have no idea how vcpkg will just fix this for everyone to mix everywhere, without patching out all of these products and many similar. You can't allocate ram using the malloc of one static CRT and then release it with a shared CRT or a different static CRT.

Not sure why we should delay adding new ports for this.

JackBoosY replied:
@BillyONeal what do you think about this?

rhuijben replied:
As far as I can tell only the Sqlite dependency currently publishes its library name. All the other dependencies don't. 
Are we now asking all libraries to start publishing their dependencies in pkg-config *or* cmake config, or?

I don't think we should delay individual packages included in this repository until everything else is converted...?

perhaps it would be easier to ask dependencies to use constant names (would just require patches to libexpat, apr and apr-util).

I don't think any of these dependencies can just fix the mixing of memory models as noted earlier. So these new configs would be broken at runtime anyway if they started using mixed crt configs.

BillyONeal replied:
>Are we now asking all libraries to start publishing their dependencies in pkg-config or cmake config, or?

It isn't "all libraries need to use pkgconfig", it is "if we have a library that needs something normally provided by pkgconfig, we should patch the source to produce pkgconfig rather than patching every consumer".

BillyONeal replied:
It looks like most of the dependencies do publish pkgconfig and there is an autoconf build system, have you considered using that @rhuijben ?

rhuijben committed:
[subversion] Patch generator to allow specifying crt as MSBuild argument

rhuijben commented:
@JackBoosY Why did you add the requires author response? I don't see new open questions. What am I missing?

JackBoosY commented:
@rhuijben Neumann have some unsolved questions.

rhuijben commented:
> @rhuijben Neumann have some unsolved questions.

Most of these would require new vcpkg features to fix.. other packages to forward info that they currently don't do... While these ports would already help users in this form.

I don't think we would like completely custom cmake scripts for every project in vcpkg that just ignore the intended builds from the projects providing the source code.

JackBoosY commented:
/azp run

JackBoosY commented on ports/serf/portfile.cmake:
```diff
@@ -0,0 +1,116 @@
+if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
+    message(FATAL_ERROR "${PORT} does not currently support UWP")
+endif()
```
```suggestion
vcpkg_fail_port_install(ON_TARGET "UWP")
```

rhuijben replied:
Thanks. Committed this fix. Not sure if this is completely necessary, as APR (required dependency) doesn't work on UWP yet.

JackBoosY commented on scripts/ci.baseline.txt:
```diff
@@ -1461,6 +1461,8 @@ sentencepiece:x86-windows=fail
 septag-sx:arm64-windows=fail
 septag-sx:arm-uwp=fail
 septag-sx:x64-uwp=fail
+serf:x64-linux=fail
+serf:x64-osx=fail
```
Is this unsupport by the upstream or temporary unsupport in vcpkg?

rhuijben replied:
For now unsupported in vcpkg. It is supported but uses a different buildsystem on these platforms. 

These Subversion packages are +- in the base system on these platforms, while they are generally hard to build on Windows. I'm trying to switch my personal build for these package to vcpkg... Maintain the patches there instead of in a separate scripted legacy buildsystem.

Getting serf to build properly most likely just requires passing a few paths in another style.
Subversion requires using a different generator and other ways to find the packages. 
Subversion uses autoconf on linux and os/x, while it mostly assumes passing all paths manually on Windows.

JackBoosY commented on scripts/ci.baseline.txt:
```diff
@@ -1543,6 +1545,8 @@ stxxl:x86-windows=skip
 stxxl:x64-windows=skip
 stxxl:x64-windows-static=skip
 stxxl:x64-windows-static-md=skip
+subversion:x64-osx=fail
+subversion:x64-linux=fail
```
Same as above.

rhuijben replied:
See answer above. Should work, but much easier to fix on a linux or mac/os system than on my windows box.

rhuijben committed:
[serf] Fix building xx-windows-static

rhuijben committed:
Add versions for serf and subversion

rhuijben commented:
@JackBoosY Fixed the serf build for x64-linux with the last version of the patch. I assume this fixes os/x as well.

JackBoosY commented:
I would like my colleague will review the PR further.

vicroms commented:
We'll review this in our Thursday meeting.

rhuijben commented:
@vicroms Any news?

JackBoosY commented:
Sorry for late, we will discuss this PR at the meeting next week.

JackBoosY commented:
Ping @vicroms @BillyONeal for response.

BillyONeal commented on ports/subversion/portfile.cmake:
```diff
@@ -0,0 +1,135 @@
+vcpkg_execute_build_process(
+    COMMAND ${PYTHON3} gen-make.py
+        -t vcproj --vsnet-version=2019 # Actual version is overridden during MSBuild step.
+        --with-apr=${CURRENT_INSTALLED_DIR}
```
It seems unlikely that these are going to correctly find the debug libraries in debug builds?

rhuijben replied:
There is magic in the python code to find both debug and release libraries. This goes through a lot of steps to generate .vcxproj files with all the right references for both debug and release builds.

The separate patches fix this version of Subversion to find them directly... And the same patches are already in Subversion trunk so should be available with the next minor release of Subversion (if not earlier)

JackBoosY commented:
Can you please merge to master?

strega-nil-ms committed:
[baseline] as of 2.0.2, portmidi:x64-windows-static-md passes (#22397)

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

gracicot committed:
[openssl] Update to 1.1.1m (#22376)

* [openssl] Update to 1.1.1m

* run x-add-version --all for openssl 1.1.1m

* switch to 4 spaces to standardize instead of 2

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

PhoebeHui committed:
[boost-modular-build-helper] Add missing dependency (#20842)

LilyWangLL committed:
[arrow] Fix CMake link error (#22387)

* [arrow] Fix CMake link error

* update version

* [arrow] update vcpkg-cmake-wrapper.cmake

* update version

Co-authored-by: Lily Wang <v-lilywang@microsoft.com>

JackBoosY committed:
[urdfdom][sdformat9] Re-fix build (#22367)

* [urdfdom][sdformat9] Re-fix build

* version

* Remove empty folder

* version

* [dartsim] Fix build issue

* version

* Update version name

* version

* Apply suggestions

* version

* Add PKG_EXPORTS

* version

* Update ports/sdformat9/fix-dependency-urdfdom.patch

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

RT2Code committed:
[plf-colony] Update to 6.33 - [plf-hive] Add new port (#22389)

* [plf-colony-hive] Update to 6.33

* [plf-hive] Add new port

* [plf-colony][plf-hive] Update version files

* [plf-colony][plf-hive] Add quotes to paths

* [plf-colony][plf-hive] Update version files

* [plf-hive] Add version file

Co-authored-by: Rémy Tassoux <rt2@rasterizedworld.com>

Adela0814 committed:
[uriparser] update to 0.9.6 (#22398)

eao197 committed:
[sobjectizer] Update to v.5.7.3 (#22399)

* sobjectizer updated to v5.7.3.

* so5extra updated to v1.5.0.

* so5extra-1.5.0 added to baseline.

* Fixes for old cmake commands.

* Update baseline for sobjectizer/so5extra.

* modernize cmake functions

* update version

* modernize cmake functions

* update version

* udpate version

Co-authored-by: Jonliu1993 <13720414433@163.com>

yurybura committed:
[boost] update to 1.78.0 (#21928)

* [scripts] update scripts

* [boost-*] generate ports

* update versions

* [scripts] update boost version in boost-modular-headers.cmake

* [scripts] add post source stubs for boost-json and boost-nowide ports and re-generate Boost ports

* update versions

* [boost-asio] make OpenSSL optional #20718

* update versions

* [boost] [boost-mpi] [boost-python] disable python for iOS and Android, fixes #21474

* update versions

* [boost-fiber] add NUMA support, fixes #16507

* update version

* [boost-build] apply patch https://github.com/bfgroup/b2/pull/113 for fix build boost-fiber and boost-stacktrace

* update version

* [boost-atomic] fix compilation for uwp

* update version

* ignore mapnik's fail on arm64-windows

* [boost-python] support ARM on Windows

* update versions

* Boost features testing within CI

* [boost-modular-build-helper] fix boost_python file name in Jamroot.jam

* [boost-mpi] fix build with python2 or python3

* update versions

* [boost-fiber] fix build with numa=on

* update version

* [boost-multiprecision] fix build with MSVC on ARM

* update version

* [boost-mpi] try to fix build on linux and osx

* update versions

* exclude static

* guarantee the same result across platforms

* minor changes to boost-modular-build-helper's CMake

* correct versions

* [boost-modular-build-helper] install Python extensions on Windows

* [boost-mpi] add patch to fix build Python extension on Windows

* update versions

* [boost-mpi] remove python2 by request @BillyONeal @strega-nil-ms

* update version

* update versions

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>
Co-authored-by: PhoebeHui <20694052+PhoebeHui@users.noreply.github.com>

rhuijben committed:
[proj4]: Make the tiff specific features configurable and don't requi… (#22402)

* [proj4]: Make the tiff specific features configurable and don't require the lossy jpeg compression that isn't used by the proj Data and/or CDN

* [proj4]: Add version

wravery committed:
[cppgraphqlgen] Update from v4.0.0 to v4.1.0 (#22411)

* Bump version from 4.0.0 to 4.1.0

* vcpkg x-add-version cppgraphqlgen

autoantwort committed:
[mman] only windows (#22453)

autoantwort committed:
[vlpp] fix mac build (#22452)

csw committed:
[s2n] disable -Werror for gcc 11 build (#22446)

s2n fails to build with gcc 11 (see
https://github.com/aws/s2n-tls/issues/2989) due to problematic type
declarations which generate compiler warnings, coupled with the fact
that s2n builds with -Werror by default. The underlying bug is not yet
fixed upstream.

Building with -Werror is not helpful for an end user trying to use s2n
from vcpkg, since a build failure like this is not actionable. This
change therefore uses the UNSAFE_TREAT_AS_ERRORS=OFF option to disable
-Werror.

fwannmacher committed:
[sdl2-image] Support x64-uwp (#22437)

* fixed 22434

* Clean baseline

* update version

* version

* add supports field since the upstream doesn't support uwp

* improve code

* version

* version

Co-authored-by: JackBoosY <yuzaiyang@beyondsoft.com>

hmwill committed:
[openblas] Upgrade to latest version 0.03.19 (#22435)

* [#22423]: Upgrade Openblas to latest version 0.03.19

* Clean-up ports/openblas. Removing patch no longer needed

* Formatting ports/openblas/vcpkg.json

* vcpkg x-add-version openblas

autoantwort committed:
[simbody] no absolute paths (#22420)

strega-nil committed:
[macos ci] update to 2022-01-03 (#22314)

* [macos ci] update to 2022-01-03

* update the instructions

* forgot to update azure-pipelines-osx.yml

* install parallels in Install-Prerequisites.ps1

* oops

* fix ace, libsoundio, and libpqxx

* tensorflow doesn't yet support macOS 12

dg0yt committed:
[drogon] Fix libmariadb import/export and usage (#22426)

* Add canonical usage

* Use unofficial-libmariadb

* Update versions

Cheney-W committed:
[getdns] Disable install COPYING in cmakelist (#22475)

Co-authored-by: Cheney-Wang <v-xincwa@microsoft.com>

SunBlack committed:
[pybind11] Update to 2.9.0 (#22460)

Co-authored-by: Heiko Thiel <heiko.thiel@hpi.de>

autoantwort committed:
[wintoast] only windows (#22441)

Pospelove committed:
[chakracore] Add flag to use all available CPUs (#22439)

* Update portfile.cmake

* Update vcpkg.json

* Update baseline.json

* Update chakracore.json

* Update portfile.cmake

* Update chakracore.json

* Update portfile.cmake

* Update portfile.cmake

* Update chakracore.json

* Update portfile.cmake

* Update chakracore.json

* Update ports/chakracore/vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

* Update chakracore.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

autoantwort committed:
[luasocket] only work on windows (#22432)

dg0yt committed:
[vcpkg_execute_build_process] Catch ld signal 9 (#22417)

* [vcpkg_execute_build_process] Catch ld signal 9

* Update message output

Ace314159 committed:
[realsense2] Update to 2.50.0 (#22416)

* Update realsense2

* Update versions

* strega-nil CRs

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

dg0yt committed:
[pdal] Revise and update (#21275)

* Modernize portfile, use vcpkg_from_github

* Revise cleanup and tool installation

* Fix static linkage with gdal

* Fix static linkage with PostgreSQL

* Fix compiler options

* Fix APPLE linker flags

* Don't use /usr/local/include

* Remove pdal failure from CI baseline

* Remove outdated FindICONV.cmake

* Simplify geos package lookup

* Separate boost patch from other dependencies

* Simplify boost patching

* Update to 2.3.0

* PDAL supports only dynamic linkage

* Normalize find_library

* Dependencies, features and plugins

* Update pdalboost fixup

* Fixup pkgconfig

"Requires" in pc file is incomplete, and not needed for dynamic linkage in vcpkg.

* Override absolute plugin path

* Don't set RPATH

* Add usage

* [pdal-c] Update portfile

* [pdal-c] Don't enforce pdal features

* [pdal-c] Update to v2.1.0

* Update versions

* Convert tabs

* Update versions

* Patch NOMINMAX definition

* Update versions

* Use project include instead of patch

* Update versions

* Update 'supports' field

* Update versions

mkhon committed:
[vcpkg] z_vcpkg_get_cmake_vars(): Use build-type-specific cache variable if VCPKG_BUILD_TYPE is defined (#22461)

* z_vcpkg_get_cmake_vars(): Use build-type-specific cache variable
if VCPKG_BUILD_TYPE is defined

Some ports (e.g. gettext) execute debug and release builds manually
(with different options). If a globally cached copy of cmake vars file
is used, the second build type has no cmake vars defined (e.g. CFLAGS is empty)
because generated and globally cached cmake vars file includes only cmake vars
for the first build type

* Apply suggestions from code review

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

Co-authored-by: nicole mazzuca <83086508+strega-nil-ms@users.noreply.github.com>

azure-sdk committed:
[azure-identity-cpp] Update to 1.1.1 (#22493)

## 1.1.1 (2022-01-11)

### Bugs Fixed

- [2741](https://github.com/Azure/azure-sdk-for-cpp/issues/2741) Fixed linking problem when Azure SDK is built as DLL.

PhoebeHui committed:
[opencolorio/lcms] Fix dependency (#21960)

FrankXie05 committed:
[NCCL] Fix usage, add wrapper and install Find module file (#22031)

* fix-find_ncclConfig.cmake

* update version

* change port-version

* update version

* update version

* update version

* update port version

* update version

Co-authored-by: Frank-Xie <Frank-Xie@Email.com>

autoantwort committed:
[libuuid] disallow on osx (#22306)

The `uuid/uuid.h` header conflicts with system libraries on osx; some system headers include `<uuid/uuid.h>`, intending to get the macOS system headers.

fairlyblank committed:
[protobuf] fix protobuf-3.18 mingw build failure (#22369)

* fix protobuf-3.18 mingw build

* fix protobuf-3.18 mingw build

yurybura committed:
[boost-modular-headers] download the license file only once (#22462)

* [boost-modular-headers] download license file only once for every header-only library

* update version

AenBleidd committed:
[wxwidgets] Fix debug-/release-only build (#22464)

Signed-off-by: Vitalii Koshura <lestat.de.lionkur@gmail.com>

LilyWangLL committed:
[buck-yeh-bux] Fix error C7595 (#22471)

* [buck-yeh-bux] Fix error C7595

* update version

Co-authored-by: Lily Wang <v-lilywang@microsoft.com>

veggiesaurus committed:
[Starlink-ast] update to 9.2.7 (#22382)

* updated to v9.2.6

* git hash update

* update to AST 9.2.7

* hash update

* Update versions/s-/starlink-ast.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

* starlink-ast doesn't use semver

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>
Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

daschuer committed:
[fontconfig] don't build font cache in case of crosscompile (#22483)

strega-nil committed:
[netcdf-c] fix installation on windows when hdf5[parallel] is installed (#22488)

fixes #21646

mkhon committed:
[gettext] Fix uwp "tools" build (#22489)

scovetta committed:
[many ports] Update HTTP to HTTPS links where it can be done safely. (#20464)

* Update HTTP to HTTPS links where it can be done safely.

This means that the HTTP-based link gave a 200 response with identical
content to the HTTPS version of the link. For links containing
variables, we checked the interpolated (real) link.

* Bump relevant vcpkg.json versions.

* Use version-date for cello.

* update version

Co-authored-by: Billy Robert O'Neal III <bion@microsoft.com>
Co-authored-by: Jonliu1993 <13720414433@163.com>

autoantwort committed:
[vtk] no absolute paths (#22330)

asklar committed:
[CppXaml] Add new port (#22414)

* Add CppXaml package

* format

* version

* 0.0.16

* update version

* Update ports/cppxaml/vcpkg.json

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>

* update version

* add double quotes

* update version

Co-authored-by: JonLiu1993 <63675417+JonLiu1993@users.noreply.github.com>
Co-authored-by: Jonliu1993 <13720414433@163.com>

niyue committed:
[antlr4] update to 4.9.3 (#22438)

JackBoosY commented:
Can you handle the osx failure? See logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=65506&view=artifacts&pathAsName=false&type=publishedArtifacts).

Honeybunch committed:
Fixing Tracy Port (#22510)

* Adding tracy port

* Formatting tracy port

* Tidying up the port

* Formatting again

* Making sure git-tree is correct

* Adding correct vcpkg dependencies

* Newline at file end again

* Update version database

* Making sure to mark tracy as not supporting x86 at all

* Updating version again

* Responding to feedback and double checking which builds will work

* Updating versions

* Tidying up some loose ends

* Updating port version again

* Removing port-version 1

* Removing tracy from ci baseline

* Removing unnecessary include copy

* Updating version

* Resolving feedback

* Updating version and formatting

* Version for the version god

* tweaking tracy supports

* Formatting and version

* Version

* Adding pthreads dependency to tracy for non-windows platforms

* Version

* Fixing tracy port

* Fixing version

* TRACY_IMPORTS should be an INTERFACE define

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

RT2Code committed:
[openxr-loader] Update to 1.0.22 (#22512)

* [openxr-loader] Update to 1.0.22

* [openxr-loader] Update version files

Co-authored-by: Rémy Tassoux <rt2@rasterizedworld.com>

Adela0814 committed:
[libmicrohttpd] update to v0.9.75 (#22514)

* update libmicrohttpd to v0.9.75

* delete useless patch file

Co-authored-by: Mengna-Li <you@example.com>

daschuer committed:
[speex] fix build on osx by disableing example binaries (#22458)

* [speex] disable building of example programs to avoid libogg dependency

* [speex] don't use deprecated vcpkg functions

PhoebeHui committed:
[vcpkg_find_fortran] Update the mingw-w64-i686-mpc* (#22519)

* [vcpkg_find_fortran] Update the mingw-w64-i686-mpc*

* Update the version for x64

PhoebeHui committed:
[icu] Fix pc files (#22388)

* [icu] Fix pc files

* Patch the issue in configure.ac

BillyONeal committed:
Update VMs for Patch Tuesday and fix pwsh CVE. (#22494)

JackBoosY commented:
Oh that's a disaster, you wrongly merged the commits.

JackBoosY commented:
Convert this PR to draft since there is no progress in a long time.

JackBoosY commented:
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.
